internal/kafka/internal/services/cloud_providers_test.go

```diff
@@ -485,10 +498,51 @@ func Test_ListCloudProviderRegions(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "failed to get provider implementation",
+			fields: fields{
+				connectionFactory: db.NewMockConnectionFactory(nil),
+				providerFactory: &clusters.ProviderFactoryMock{
+					GetProviderFunc: func(providerType api.ClusterProviderType) (clusters.Provider, error) {
+						return nil, errors.New("failed to get provider implementation")
+					},
+				},
+			},
+			wantErr: true,
+			setupFn: func() {},
+		},
+		{
+			name: "failed to retrieve cloud regions",
```

Comment:
Same comment as https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/981/files#r862961723 but adapted to cloud regions